test/test_source_tree_rewriter.rb

```diff
@@ -260,4 +265,91 @@ def test_merge
       [:wrap, @hello.join(@world), '@', '@'],
     ])
   end
+
+  def representation_example
+    Parser::Source::TreeRewriter.new(@buf)
+      .wrap(range(1...10), '(', ')')
+      .insert_after(range(2...6), '!')
+      .replace(range(2...4), 'foo')
+      .remove(range(5...6))
+  end
+
+  def test_nested_actions
+    result = representation_example.as_nested_actions
+
+    assert_equal( [ [:wrap, 1...10, '(', ')'],
+                    [:wrap, 2...6, '', '!'],  # aka "insert_after"
+                    [:replace, 2...4, 'foo'],
+                    [:replace, 5...6, ''],  # aka "removal"
+                  ],
+                  result.each {|arr| arr[1] = arr[1].to_range }
+                )
+  end
+
+  def test_ordered_replacements
+    result = representation_example.as_replacements
+
+    assert_equal( [ [ 1...1, '('],
+                    [ 2...4, 'foo'],
+                    [ 5...6, ''],
+                    [ 6...6, '!'],
+                    [ 10...10, ')'],
+                  ],
+                  result.map {|r, s| [r.to_range, s]}
+                )
+  end
+end
+
+class TestSourceTreeRewriterMerge < Minitest::Test
```

Comment:
I'd probably inherit `TestSourceTreeRewriterMerge` from `TestSourceTreeRewriter` (or there could be a single base class). Dependencies between test classes are unusual.

Reply:
I don't know Minitest well (and don't intend to, I'm too much a fan of rspec), but that's what I did first and the tests of `TestSourceTreeRewriter` get executed twice...

Reply:
If you prefer I could simply add the extra setup to the first part. I miss RSpec!

Reply:
Right, minitest simply invokes all `/test_.*/` methods that are defined twice because of the inheritance (I'm sure it could to `instance_methods(false)` to drop inherited methods, no idea why it's designed this way). However, creating a base class with a `setup` method (and inheriting 2 test classes from it) could work. Let's keep it as is